### ISSUE TITLE
Pull Policy Error

### ISSUE THREAD

**Issue by kanowah** [telekom-security/tpotce#1623]
When I'm trying to install the new version, I'm getting the following error, I've tried to installed it on different devices still same:

![Screenshot 2024-07-23 144016](https://github.com/user-attachments/assets/cfea0715-98fe-441f-8d55-e81af96e204c)

**Comment by t3chn0m4g3** [comment]
Check the `docker-compose.yml`, it looks like you made changes around the `map_web` service and now T-Pot cannot start due to a missing setting.

**Comment by kanowah** [comment]
It's a fresh installation, I have not change any file at all

**Comment by kanowah** [comment]
I've only used the following command:

env bash -c "$(curl -sL https://github.com/telekom-security/tpotce/raw/master/install.sh)"

**Comment by t3chn0m4g3** [comment]
Then post the `docker-compose.yml` as formatted markdown because I cannot reproduce it and provide infos about your OS and such.

**Comment by t3chn0m4g3** [comment]
Thanks for the update, I can now reproduce it with a completely fresh install on Debian 12.
It looks like a new docker version is the [culprit](https://github.com/docker/compose/issues/12001).

**Comment by mygitcrazy** [comment]
I also spinned up a fresh installation a couple of days ago and it worked instantly. Today i installed the same installation again, and also getting the error as posted in this thread....

Ubuntu 24.04 and tried Debian 12 (same error output!)

**Comment by t3chn0m4g3** [comment]
A possible workaround...

Run `sudo apt-cache madison docker-compose-plugin` and it will spit out all the available repo versions:
```
docker-compose-plugin | 2.29.0-1~debian.12~bookworm | https://download.docker.com/linux/debian bookworm/stable arm64 Packages
docker-compose-plugin | 2.28.1-1~debian.12~bookworm | https://download.docker.com/linux/debian bookworm/stable arm64 Packages
[...]
```

We are only interested in the `2.28` branch and you can now install it with...
`sudo apt-get install docker-compose-plugin=2.28.1-1~debian.12~bookworm`

Then goto `$HOME/tpotce` and run `docker compose pull`.

Let me know if this works and I will update this for other OS installs if requested.

**Comment by mygitcrazy** [comment]
Up and running 🎉
This (workaround) did work for me!!

**Comment by t3chn0m4g3** [comment]
Thanks for the quick feedback @mygitcrazy! Happy it is working 🤩

**Comment by akajhon** [comment]
Works like a charm! Thanks for the fast help @t3chn0m4g3 !

**Comment by t3chn0m4g3** [comment]
@akajhon Appreciate the fast response @akajhon! Happy it is working for you, too! 🤩

**Comment by t3chn0m4g3** [comment]
I will monitor this and as soon as Docker releases a fixed version I will update this issue accordingly.

Thanks @mygitcrazy @akajhon @kanowah bringing this up and testing the workaround.

**Comment by t3chn0m4g3** [comment]
The upstream issue has been closed and the fix is available in the [2.29.1](https://github.com/docker/compose/releases/tag/v2.29.1) release. No details in the description, though and package not yet pushed to Docker Linux repositories.

**Comment by t3chn0m4g3** [comment]
The new docker-compose-plugin was pushed to the Docker Linux repositories, just tested it and I can confirm this is resolved now.
